lib/src/Pos/Client/CLI/NodeOptions.hs

```diff
@@ -125,17 +126,15 @@ data NodeArgs = NodeArgs
     } deriving Show
 
 simpleNodeArgsParser :: Parser SimpleNodeArgs
-simpleNodeArgsParser = do
-    commonNodeArgs <- commonNodeArgsParser
-    behaviorConfigPath <- behaviorConfigOption
-    pure $ SimpleNodeArgs commonNodeArgs NodeArgs{..}
-
-behaviorConfigOption :: Parser (Maybe FilePath)
-behaviorConfigOption =
-    optional $ strOption $
-        long "behavior" <>
-        metavar "FILE" <>
-        help "Path to the behavior config"
+simpleNodeArgsParser = SimpleNodeArgs
+    <$> commonNodeArgsParser
+    <*> nodeArgsParser
+
+nodeArgsParser :: Parser NodeArgs
+nodeArgsParser = fmap NodeArgs $ optional $ strOption $
+    long "behavior" <>
+    metavar "FILE" <>
+    help "Path to the behavior config"
```

Comment:
Don't really understand why this wasn't done this way at first. But I need the parser to extract available arguments in the `demo` package. Hence the small refactor to expose the parser and actually makes the intent a bit clearer.